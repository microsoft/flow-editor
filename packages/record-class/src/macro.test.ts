import plugin from "babel-plugin-macros";
import pluginTester from "babel-plugin-tester";

pluginTester({
  plugin,
  pluginName: "macro-record-class",
  filename: __filename,
  babelOptions: {
    plugins: [["@babel/plugin-syntax-decorators", { legacy: true }]],
    presets: [
      [
        "@babel/preset-typescript",
        {
          allowDeclareFields: true,
        },
      ],
    ],
  },
  snapshot: true,
  tests: [
    {
      title: "simple",
      fixture: "./__fixtures__/1.ts",
    },
    {
      title: "ignore fields that are not public readonly",
      fixture: "./__fixtures__/2.ts",
    },
  ],
});
