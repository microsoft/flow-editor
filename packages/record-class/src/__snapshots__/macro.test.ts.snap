// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macro-record-class ignore fields that are not public readonly: ignore fields that are not public readonly 1`] = `

import { RecordBase } from "../lib/record";
import record from "./proxy.macro";

@record
export class SomeRecord extends RecordBase<SomeRecord, SomeRecord> {
  public field1?: string = "";
  private readonly field2 = false;

  public override $$create(partial: Partial<SomeRecord>): SomeRecord {
    return new SomeRecord(partial);
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { RecordBase } from "../lib/record";
export class SomeRecord extends RecordBase {
  field1 = "";
  field2 = false;

  $$create(partial) {
    return new SomeRecord(partial);
  }

  $$initialize(partial) {}

  merge(partial) {
    return this.$$create({});
  }

  pipe(...list) {
    let next = this.clone();
    list.forEach((f) => {
      Object.assign(next, f(next));
    });
  }

  toJSON() {
    return {};
  }
}


`;

exports[`macro-record-class simple: simple 1`] = `

import { RecordBase } from "../lib/record";
import record from "./proxy.macro";

@record
export class SomeRecord extends RecordBase<SomeRecord, SomeRecord> {
  public readonly name?: string = "";

  constructor(partial: Partial<SomeRecord>) {
    super(partial);
  }

  public someMethod(): void {
    //
  }

  public override $$create(partial: Partial<SomeRecord>): SomeRecord {
    return new SomeRecord(partial);
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { RecordBase } from "../lib/record";
export class SomeRecord extends RecordBase {
  name;

  constructor(partial) {
    super(partial);
  }

  someMethod() {
    //
  }

  $$create(partial) {
    return new SomeRecord(partial);
  }

  $$initialize(partial) {
    this.name = partial.name ?? "";
  }

  merge(partial) {
    return this.$$create({
      name: partial.name ?? this.name,
    });
  }

  pipe(...list) {
    let next = this.clone();
    list.forEach((f) => {
      Object.assign(next, f(next));
    });
  }

  toJSON() {
    return {
      name: this.name,
    };
  }
}


`;
